source/postprocess/viscous_dissipation_statistics.cc

```diff
@@ -78,11 +80,19 @@ namespace aspect
             // calculate the local viscous dissipation integral
             for (unsigned int q = 0; q < n_q_points; ++q)
               {
+                // In some material models a part of the total work done is
+                // converted into other energy types than heat, e.g. surface energy.
+                // If we use such a material model, reduce the work fraction appropriately.
+                double heating_work_fraction = 1.0;
+                if (const MaterialModel::DislocationViscosityOutputs<dim> *disl_viscosities_out =
+                      out.template get_additional_output<MaterialModel::DislocationViscosityOutputs<dim> >())
+                  heating_work_fraction -= disl_viscosities_out->boundary_area_change_work_fractions[q];
```

Comment:
This modification is in one of the output postprocessors. Would we have to do similar modifications in any other parts of the code? It's a bit unfortunate that we would need to adjust *every single place* of the code base with knowledge about one particular additional heating model thingy...